Makefile

```diff
@@ -126,6 +127,7 @@ include $(MAKE_SCRIPT_DIR)/config.mk
 ifeq ($(CONFIG),)
 ifeq ($(TARGET),)
 TARGET := $(DEFAULT_TARGET)
+SKIPCHECKS := yes
```

Comment:
Is this necessary? IMO this PR can default to check

Reply:
The issue is if you run `make` on its own it does the check for the default target. The effect of this is it may fail for the default target - but not for the target you specify (first iteration of make).

e.g. 
```
make       <- runs STM32F405
make STM32F7X2                   <- runs as default target context, then as TARGET=STM32F7X2 - results in possible errors unrelated to the target (or config) the user is running for
make TARGET=STM32F7X2     <- works fine, as the target is set directly.
```

We could remove the default target, and force `make X` or `make TARGET=X`

Reply:
and running test only when TARGET is set(second invocation)?

Reply:
> and running test only when TARGET is set(second invocation)?

Exactly - but TARGET is always set - because we want `make` to simply build the STM32F405. If we remove that (which I am not against) then there is no issue, and the skip check on default target can be removed.

Reply:
- removal of a default is okay by me
- i never specify `TARGET=X` but rather only use X
- much as running any linux command, make could spit out usage if nothing specified. if nothing else, print at least `"make <tab><tab>" for available options`.